### ISSUE TITLE
Encountered error "Bad state: Empty input given" while using flutter for web

### ISSUE THREAD

**Issue by ashutoshsingh05** [dart-lang/sdk#39745]
Everything was working fine until I changed one theme parameter in my app, tried to do a hot restart and received the below error message.

```
We're sorry, you've found a bug in our compiler.
You can report this bug at:
    https://github.com/dart-lang/sdk/issues/labels/web-dev-compiler
Please include the information below in your report, along with
any other information that may help us track it down. Thanks!
-------------------- %< --------------------
    dartdevc -k arguments: --dart-sdk-summary=/home/ashutosh/flutter/bin/cache/flutter_web_sdk/kernel/flutter_ddc_sdk.dill --modules=amd --no-summarize -o
    packages/personal_website/main_web_entrypoint.ddc.js --source-map --summary=packages/js/js_util.ddc.dill=packages/js/js_util --summary=packages/js/js.ddc.dill=packages/js/js
    --summary=packages/firebase/src/app.ddc.dill=packages/firebase/src/app --summary=packages/firebase/firebase.ddc.dill=packages/firebase/firebase
    --summary=packages/meta/meta.ddc.dill=packages/meta/meta --summary=packages/typed_data/typed_buffers.ddc.dill=packages/typed_data/typed_buffers
    --summary=packages/flutter/src/foundation/_bitfield_web.ddc.dill=packages/flutter/src/foundation/_bitfield_web
    --summary=packages/flutter/foundation.ddc.dill=packages/flutter/foundation --summary=packages/vector_math/hash.ddc.dill=packages/vector_math/hash
    --summary=packages/vector_math/vector_math_64.ddc.dill=packages/vector_math/vector_math_64
    --summary=packages/flutter/src/physics/clamped_simulation.ddc.dill=packages/flutter/src/physics/clamped_simulation --summary=packages/flutter/physics.ddc.dill=packages/flutter/physics
    --summary=packages/collection/src/utils.ddc.dill=packages/collection/src/utils --summary=packages/collection/src/algorithms.ddc.dill=packages/collection/src/algorithms
    --summary=packages/collection/src/iterable_zip.ddc.dill=packages/collection/src/iterable_zip --summary=packages/collection/src/comparators.ddc.dill=packages/collection/src/comparators
    --summary=packages/collection/src/priority_queue.ddc.dill=packages/collection/src/priority_queue
    --summary=packages/collection/src/canonicalized_map.ddc.dill=packages/collection/src/canonicalized_map --summary=packages/collection/collection.ddc.dill=packages/collection/collection
    --summary=packages/flutter/src/gestures/arena.ddc.dill=packages/flutter/src/gestures/arena --summary=packages/flutter/gestures.ddc.dill=packages/flutter/gestures
    --summary=packages/flutter/src/painting/_network_image_web.ddc.dill=packages/flutter/src/painting/_network_image_web
    --summary=packages/flutter/painting.ddc.dill=packages/flutter/painting --summary=packages/flutter/src/semantics/binding.ddc.dill=packages/flutter/src/semantics/binding
    --summary=packages/flutter/semantics.ddc.dill=packages/flutter/semantics --summary=packages/flutter/src/animation/animation.ddc.dill=packages/flutter/src/animation/animation
    --summary=packages/flutter/animation.ddc.dill=packages/flutter/animation --summary=packages/flutter/src/rendering/animated_size.ddc.dill=packages/flutter/src/rendering/animated_size
    --summary=packages/flutter/rendering.ddc.dill=packages/flutter/rendering --summary=packages/flutter/src/widgets/actions.ddc.dill=packages/flutter/src/widgets/actions
    --summary=packages/flutter/widgets.ddc.dill=packages/flutter/widgets --summary=packages/flutter/src/cupertino/action_sheet.ddc.dill=packages/flutter/src/cupertino/action_sheet
    --summary=packages/flutter/cupertino.ddc.dill=packages/flutter/cupertino --summary=packages/flutter/material.ddc.dill=packages/flutter/material
    --summary=packages/generic_bloc_provider/generic_bloc_provider.ddc.dill=packages/generic_bloc_provider/generic_bloc_provider
    --summary=packages/personal_website/globalData.ddc.dill=packages/personal_website/globalData
    --summary=packages/personal_website/widgets/imageLayer.ddc.dill=packages/personal_website/widgets/imageLayer
    --summary=packages/personal_website/bloc/introEvents.ddc.dill=packages/personal_website/bloc/introEvents
    --summary=packages/personal_website/bloc/introBloc.ddc.dill=packages/personal_website/bloc/introBloc
    --summary=packages/personal_website/widgets/assetLayer.ddc.dill=packages/personal_website/widgets/assetLayer
    --summary=packages/personal_website/widgets/textLayer.ddc.dill=packages/personal_website/widgets/textLayer
    --summary=packages/personal_website/widgets/parallaxComponent.ddc.dill=packages/personal_website/widgets/parallaxComponent
    --summary=packages/personal_website/widgets/bottomBar.ddc.dill=packages/personal_website/widgets/bottomBar
    --summary=packages/personal_website/widgets/parallaxBuilder.ddc.dill=packages/personal_website/widgets/parallaxBuilder
    --summary=packages/firebase/firestore.ddc.dill=packages/firebase/firestore
    --summary=packages/personal_website/widgets/visitorCounter.ddc.dill=packages/personal_website/widgets/visitorCounter
    --summary=packages/personal_website/bloc/themeEvents.ddc.dill=packages/personal_website/bloc/themeEvents
    --summary=packages/personal_website/bloc/themeBloc.ddc.dill=packages/personal_website/bloc/themeBloc
    --summary=packages/personal_website/widgets/themeIcon.ddc.dill=packages/personal_website/widgets/themeIcon
    --summary=packages/personal_website/homepage.ddc.dill=packages/personal_website/homepage --summary=packages/personal_website/main.ddc.dill=packages/personal_website/main
    --packages=file:///tmp/kernel_builder_LCYMZC/.packages --module-name=packages/personal_website/main_web_entrypoint --multi-root-scheme=org-dartlang-app --multi-root=.
    --track-widget-creation --inline-source-map --libraries-file=file:///home/ashutosh/flutter/bin/cache/flutter_web_sdk/libraries.json
    --used-inputs-file=/tmp/ddk_builder_SXUFBP/used_inputs.txt package:personal_website/main_web_entrypoint.dart
    dart --version: 2.7.0-dev.2.1.flutter-8b8894648f (Mon Dec 9 13:45:47 2019 +0000) on "linux_x64"

Bad state: Empty input given.
#0      BinaryBuilder._checkEmptyInput (package:kernel/binary/ast_from_binary.dart:434:29)
#1      BinaryBuilder.readComponent.<anonymous closure> (package:kernel/binary/ast_from_binary.dart:451:7)
#2      Timeline.timeSync (dart:developer/timeline.dart:163:22)
#3      BinaryBuilder.readComponent (package:kernel/binary/ast_from_binary.dart:449:21)
#4      ProcessedOptions.loadComponent (package:front_end/src/base/processed_options.dart:410:10)
#5      initializeIncrementalCompiler (package:front_end/src/api_unstable/modular_incremental_compilation.dart:206:33)
<asynchronous suspension>
#6      initializeIncrementalCompiler (package:front_end/src/api_unstable/ddc.dart:163:10)
#7      _compile (package:dev_compiler/src/kernel/command.dart:280:27)
#8      compile (package:dev_compiler/src/kernel/command.dart:45:18)
#9      compile (package:dev_compiler/src/compiler/shared_command.dart:404:12)
#10     _CompilerWorker.performRequest.<anonymous closure> (file:///b/s/w/ir/cache/builder/src/third_party/dart/pkg/dev_compiler/bin/dartdevc.dart:64:13)
#11     _rootRun (dart:async/zone.dart:1126:13)
#12     _CustomZone.run (dart:async/zone.dart:1023:19)
#13     _runZoned (dart:async/zone.dart:1518:10)
#14     runZoned (dart:async/zone.dart:1465:12)
#15     _CompilerWorker.performRequest (file:///b/s/w/ir/cache/builder/src/third_party/dart/pkg/dev_compiler/bin/dartdevc.dart:62:24)
#16     AsyncWorkerLoop.run.<anonymous closure> (package:bazel_worker/src/worker/async_worker_loop.dart:33:41)
#17     _rootRun (dart:async/zone.dart:1126:13)
#18     _CustomZone.run (dart:async/zone.dart:1023:19)
#19     _runZoned (dart:async/zone.dart:1518:10)
#20     runZoned (dart:async/zone.dart:1465:12)
#21     AsyncWorkerLoop.run (package:bazel_worker/src/worker/async_worker_loop.dart:33:26)
<asynchronous suspension>
#22     main (file:///b/s/w/ir/cache/builder/src/third_party/dart/pkg/dev_compiler/bin/dartdevc.dart:28:57)
#23     _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:303:32)
#24     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:174:12)
```

Here's the output of flutter doctor -v
```
[✓] Flutter (Channel master, v1.13.1-pre.94, on Linux, locale en_IN)
    • Flutter version 1.13.1-pre.94 at /home/ashutosh/flutter
    • Framework revision 36e599eb5d (11 hours ago), 2019-12-11 07:27:13 +0100
    • Engine revision 12bf95fd49
    • Dart version 2.7.0 (build 2.7.0-dev.2.1 8b8894648f)

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /home/ashutosh/Android/Sdk
    • Android NDK location not configured (optional; useful for native profiling
      support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /home/ashutosh/Downloads/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Android Studio (version 3.5)
    • Android Studio at /home/ashutosh/Downloads/android-studio
    • Flutter plugin version 41.0.2
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.40.2)
    • VS Code at /usr/share/code
    • Flutter extension version 3.7.1

[✓] Connected device (2 available)
    • Chrome     • chrome     • web-javascript • Google Chrome 79.0.3945.79
    • Web Server • web-server • web-javascript • Flutter Tools

• No issues found!

```

---
EDIT 1
Upon subsequent rebuilds, I get an error message stating that if this error is persistent, I should look over to #38102 

---
EDIT 2
Running `flutter clean` has resolved the issue for now and I'm able to build successfully.

**Comment by jakemac53** [comment]
@vsmenon This looks like our first report since adding in the retry logic - so it hasn't (at least fully) solved the problem.

**Comment by jakemac53** [comment]
I am going to close this as a duplicate of #38102, but thank you for the report. We were hoping this would be resolved in 2.7 but clearly it isn't, and its good to know that :).
